### PR TITLE
Migrate dialog to two stimulus controller, DialogController and teleportController

### LINKED ISSUE #10174: 🎛️ Migrate `includes/dialog` to a Stimulus Controller `w-dialog`
> ℹ️ **Part of the [Stimulus 🎛️ RFC 78](https://github.com/wagtail/rfcs/pull/78)**

### Is your proposal related to a problem?

The current `dialog` implementation is relatively simple and uses `a11y-dialog` under the hood. The wrapper is written in JavaScript and it would be great to see this move to Stimulus with adoption of TypeScript.

### Describe the solution you'd like

See the current implementation in the following files

* [client/src/includes/dialog.js](https://github.com/wagtail/wagtail/blob/main/client/src/includes/dialog.js)
* [wagtail/admin/templates/wagtailadmin/shared/dialog/dialog.html](https://github.com/wagtail/wagtail/blob/main/wagtail/admin/templates/wagtailadmin/shared/dialog/dialog.html)

**Proposed implementation**

> **IMPORTANT:** Read comments, this approach has been refined further down with the creation of TWO controllers.

We create one new controller `DialogController`/`w-dialog` that has two main modes, the first is the template mode, and the second is the dialog mode.

The template mode will only trigger if there is a `template` target, this will do the bulk of the work in `client/src/includes/dialog.js` of creating a DOM copy of the content, appending it the selected root container (or body).

The dialog mode however, will do the instantiation to `A11yDialog`, when there is no template target. The existing declared show/hide event listeners no longer need to be declared in JS but instead can be declared on the HTML elements via data attributes.

The new controller will need to have unit tests to support the behaviour.

Finally, the `body` content will be used as a target and other selectors to be updated to align with this.

```html
{% load wagtailadmin_tags i18n %}
<template
    data-wagtail-dialog
    data-controller="w-dialog"
    data-w-dialog-target="template"
    {% if dialog_root_selector %}data-w-dialog-root-selector-value="{{ dialog_root_selector }}"{% endif %}
>
    <div
        id="{{ id }}"
        aria-labelledby="title-{{ id }}"
        aria-hidden="true"
        class="w-dialog {% if theme %}w-dialog--{{ theme }}{% endif %} {% if classname %} {{ classname }}{% endif %}"
        data-controller="w-dialog"
        data-action="w-dialog:hide->w-dialog#hide w-dialog:show->w-dialog#show"
        {% if data_url %}data-url="{{ data_url }}"{% endif %}
        {% if theme %}data-w-dialog-theme-value="{{ theme }}"{% endif %}
    > 
    {% comment %} ... all the other stuff - as is {% endcomment %}

                <div data-w-dialog-target="body">
                    {{ children }}
                </div>
            </div>
        </div>
    </div>
</template>
```

The new controller will have the following

* Targets; `body` and `template`
* Methods:
  * `connect` (if in template mode, calls clone, otherwise create)
  * `create` (for when in dialog mode, connects to the A11y instance, e.g. `this.a11y = A11y(this.element);`)
  * `clone` (for when connected and a template target exists)
  * `hide` - calls hide on the A11y instance
  * `show`- calls show on the A11y instance
  * `disconnect` - should call `destroy` on the A11y instance
* Values:
  * `container` (or `rootSelector`) where the dialog should be created, note that it should default to `body`. This is only 
applicable in the template mode.
  * `theme`, this is only applicable on the dialog mode, used to determine behaviour of body scroll (see JS implementation)
* Dispatched events
  * Along with the supported listeners for `w-dialog:show` & `w-dialog:hide` (added in the DOM)
  * We should also dispatch some ``w-dialog:shown` ``w-dialog:hidden` (or similar) when the actual modal does toggle, this ability needs to be registered on the A11y instantiation.

### Describe alternatives you've considered

* Leave as is, the current implementation covers most cases.

### Additional context

* We will need to ensure that the userbar can support this new approach, it will need to potentially create its own Stimulus application instance.
* We will need to ensure that any existing data attributes in use, or other event listeners/dispatching usage has the code updated.

### PR DESCRIPTION
**Pull request by Lovelyfin00**
Fixes #10174 

## Fix Summary

- Moved the dialog.js code from JavaScript to Stimulus and deleted the file
- Split the codes into two controllers. One is the DialogController and the other is the TeleportController
- DialogController is responsible for displaying the A11y dialog
- Teleport is responsible for grabbing and appending the template elements
- Kind of made it work in userbar accessibility toggle

<br />

## Working demo

[chrome-capture-2023-4-2.webm](https://user-images.githubusercontent.com/38161296/235628858-5cbfbaf5-6bf2-47bb-91ed-2f88eec8a614.webm)

[chrome-capture-2023-4-2 (1).webm](https://user-images.githubusercontent.com/38161296/235629114-8afa519c-6e2f-4a30-8cb4-4c2bbf8706b1.webm)

## Issues

- I don't know why we had to set a `data-w-dialog-target` for the body div since it's not being used in the controller code? Why not leave the previous data-attribute (`data-dialog-body`)  since that attribute is what is being grabbed from inside the userbar.ts file.
- I wasn't exactly able to grab the controller and reuse it inside the userbar so I had to import A11y and wra`p and append the template content inside of the userbar.ts file.
@lb-

### CODE REVIEW

**Comment by Lovelyfin00** [comment]
@lb- 
After fixing the conflicts
I get this error in my console
image

**Comment by Lovelyfin00** [comment]
> > @lb- After fixing the conflicts I get this error in my console
> 
> Hey. Try a fresh `npm ci` - it seems to be something odd with Typescript, webpack and the autosize package. I had the same warning and fresh npm install helped. Also double check you're on mode 18

Alright
Will do that now

File: client/src/controllers/TeleportController.ts [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+    ) as HTMLElement;
+
+    if (!this.targetContainer) {
+      throw new Error('Target container not found');
+    }
+
+    this.append();
+  }
+
+  append() {
```
Comment (lb-) [inline client/src/controllers/TeleportController.ts:34]: Split the getter out to it's own method - this makes it easier to override (see userbar code). 
```suggestion
  get targetContainer() {
      this.targetContainer = document.querySelector(
      this.targetValue,
    ) as HTMLElement;

    if (!this.targetContainer) {
      throw new Error('Target container not found');
    }
    }

  append() {
```

**Review by lb- — COMMENTED** [review]
Just some misc comments. Hope this helps.

File: wagtail/admin/templates/wagtailadmin/shared/dialog/dialog.html [hunk 2]
```diff
@@ -1,10 +1,15 @@
 {% load wagtailadmin_tags i18n %}
-<template data-wagtail-dialog>
+<template
+    data-controller="w-teleport"
+    {% if dialog_root_selector %}data-w-teleport-target-value="{{ dialog_root_selector }}"{% endif %}
+>
     <div id="{{ id }}"
+        data-controller="w-skiplink w-dialog"
```
[the remark this answers is not in the document]
Comment (Lovelyfin00) [reply wagtail/admin/templates/wagtailadmin/shared/dialog/dialog.html:7]: It's a mistake😥
I thought there was a previous skiplink controller there.

File: client/src/includes/userbar.ts [hunk 3]
```diff
@@ -1,6 +1,6 @@
 import axe, { ElementContext, NodeResult, Result, RunOptions } from 'axe-core';
 
-import { dialog } from './dialog';
+import A11yDialog from 'a11y-dialog';
```
[the remark this answers is not in the document]
Comment (Lovelyfin00) [reply client/src/includes/userbar.ts:3]: > // However, make the template controller a slightly different version that attaches to the shadowRoot

Hi @lb- 
I don't understand this part
After creating the `ShadowTemplateController` inside of userbar.ts, and creating the definitions, how would the ShadowTemplateController be different? It's already returning the targetContainer on connect.

    
    this.Stimulus = new Stimulus(shadowRoot); // ensure we only attach Stimulus to the right DOM
    const definitions= {}// teleport/dialog
    this.Stimulus.load(definitions);

Then this part too. We didn't import `Stimulus` so this won't work

**Review by lb- — COMMENTED** [review]
@Lovelyfin00  - I have done a bit of testing / playing with this code locally. It is looking great, I have also been able to validate the idea of putting a Stimulus application inside the userbar code and have got that working.

Here are the items that need some work, at the end of this comment I will put some code snippets.

You do not have to use all this code but hopefully it helps give a bit of direction here.

### 1. `wagtail/admin/templates/wagtailadmin/userbar/base.html`

* This `template` has two elements inside, two divs. The file will need to be updated to just render ONE root div - this will make the code much more reliable and easier to work with.

See below, the added `div` under `template` (closing div needed also + re-format).

```html
<!-- Wagtail user bar embed code -->
<template id="wagtail-userbar-template">
<div>
    <div class="w-userbar w-userbar--{{ position|default:'bottom-right' }}" data-wagtail-userbar part="userbar">
```


### 2. `wagtail/admin/templates/wagtailadmin/shared/dialog/dialog.html`

* `data-dialog-root-selector="{{ dialog_root_selector }}"` this code needs to move to the `template` and not on the dialog div.
* We will also need to still support the `wagtail:hide` and `wagtail:show` events until Wagtail 6.0.

Proposed HTML

```html
{% load wagtailadmin_tags i18n %}
<template
    data-controller="w-teleport"
    data-w-teleport-target-value="{% if dialog_root_selector %}{{ dialog_root_selector }}{% else %}{% endif %}"
>
    <div id="{{ id }}"
        aria-labelledby="title-{{ id }}"
        aria-hidden="true"
        class="w-dialog{% if theme %} w-dialog--{{ theme }} {% endif %}{% if classname %} {{ classname }}{% endif %}"
        data-controller="w-dialog"
        {% comment %} RemovedInWagtail60 - wagtail:hide & wagtail:show events to be removed (last two below) {% endcomment %}
        data-action="w-dialog:hide->w-dialog#hide w-dialog:show->w-dialog#show wagtail:hide->w-dialog#hide wagtail:show->w-dialog#show"
        {% if theme %}data-w-dialog-theme-value="{{ theme }}"{% endif %}
        {% if data_url %}data-url="{{ data_url }}"{% endif %}
    >
        <div data-a11y-dialog-hide class="w-dialog__overlay"></div>
```

### 3. `client/src/entrypoints/admin/modal-workflow.js`

- A reminder to use the new event names (line 134)

```ts
      self.dialog.dispatchEvent(new CustomEvent('w-dialog:hide'));
```

### 4. Dialog Controller & tests

* Move the JSDoc to the controller, not on the connect method, reworded slightly
* Ensure the body is an actual target
* Dispatch some useful events when main modal things happen
* Add show/hide methods
* A starting point for unit tests - hope this helps

```ts
import { Controller } from '@hotwired/stimulus';
import A11yDialog from 'a11y-dialog';

/**
 * Instantiates A11y dialog on hide and show events
 *
 * @example
 * <div
 *   data-controller="w-dialog"
 *   data-w-dialog-theme-value="floating"
 * >
 *   <div data-w-dialog-target="body">CONTENT</div>
 * </div>
 */
export class DialogController extends Controller<HTMLElement> {
  static values = {
    theme: { default: '', type: String },
  };

  static targets = ['body'];

  declare dialog: A11yDialog;
  declare readonly bodyTarget: HTMLElement;
  declare readonly themeValue: string;

  connect() {
    const isFloating = this.themeValue === 'floating';
    this.dialog = new A11yDialog(this.element);
    const detail = { body: this.bodyTarget, dialog: this.dialog };

    this.dispatch('ready', { detail });

    this.dialog
      .on('show', () => {
        if (!isFloating) document.documentElement.style.overflowY = 'hidden';

        this.dispatch('shown', { detail });
      })
      .on('hide', () => {
        if (!isFloating) document.documentElement.style.overflowY = '';

        this.dispatch('hidden', { detail });
      });
  }

  hide() {
    this.dialog.hide();
  }

  show() {
    this.dialog.show();
  }
}
```

```ts
import { Application } from '@hotwired/stimulus';

import A11yDialog from 'a11y-dialog';
import { DialogController } from './DialogController';

describe('DialogController', () => {
  let application;

  describe('basic behaviour', () => {
    beforeEach(() => {
      application?.stop();

      document.body.innerHTML = `
  <section>
    <div
      id="dialog-container"
      aria-hidden="true"
      data-controller="w-dialog"
      data-action="w-dialog:hide->w-dialog#hide w-dialog:show->w-dialog#show"
    >
      <div role="document">
        <div id="dialog-body" data-w-dialog-target="body">CONTENT</div>
      </div>
    </div>
  </section>`;
      application = new Application();
      application.register('w-dialog', DialogController);
    });

    it('should instantiate the controlled element with the A11y library', async () => {
      const listener = jest.fn();
      document.addEventListener('w-dialog:ready', listener);

      expect(listener).not.toHaveBeenCalled();
      application.start();

      await Promise.resolve();

      expect(listener).toHaveBeenCalled();
      const { body, dialog } = listener.mock.calls[0][0].detail;

      expect(body).toEqual(document.getElementById('dialog-body'));
      expect(dialog).toBeInstanceOf(A11yDialog);
      expect(dialog.$el).toEqual(document.getElementById('dialog-container'));
    });

    it('should support the ability to show and hide the dialog', async () => {
      const shownListener = jest.fn();
      document.addEventListener('w-dialog:shown', shownListener);

      const hiddenListener = jest.fn();
      document.addEventListener('w-dialog:hidden', hiddenListener);

      application.start();

      await Promise.resolve();

      expect(shownListener).not.toHaveBeenCalled();
      expect(hiddenListener).not.toHaveBeenCalled();
      const dialog = document.getElementById('dialog-container');

      // closed by default
      expect(dialog.getAttribute('aria-hidden')).toEqual('true');
      expect(document.documentElement.style.overflowY).toBe('');

      // show the dialog manually
      dialog.dispatchEvent(new CustomEvent('w-dialog:show'));

      expect(dialog.getAttribute('aria-hidden')).toEqual(null);
      expect(shownListener).toHaveBeenCalledWith(
        expect.objectContaining({
          detail: expect.objectContaining({
            body: document.getElementById('dialog-body'),
            dialog: expect.any(Object),
          }),
        }),
      );
      expect(hiddenListener).not.toHaveBeenCalled();
      // add style to root element on shown by default
      expect(document.documentElement.style.overflowY).toBe('hidden');

      // hide the dialog manually
      dialog.dispatchEvent(new CustomEvent('w-dialog:hide'));

      expect(dialog.getAttribute('aria-hidden')).toEqual('true');
      expect(shownListener).toHaveBeenCalledTimes(1);
      expect(hiddenListener).toHaveBeenCalledWith(
        expect.objectContaining({
          detail: expect.objectContaining({
            body: document.getElementById('dialog-body'),
            dialog: expect.any(Object),
          }),
        }),
      );
      // reset style on root element when hidden by default
      expect(document.documentElement.style.overflowY).toBe('');
    });

    // TODO
    it.skip('should support the ability use a theme to avoid document style change', async () => {
      // update the DOM with the theme value
      application.start();

      await Promise.resolve();

      const dialog = document.getElementById('dialog-container');

      // closed by default
      expect(dialog.getAttribute('aria-hidden')).toEqual('true');

      dialog.dispatchEvent(new CustomEvent('w-dialog:show'));

      expect(dialog.getAttribute('aria-hidden')).toEqual(null);

      dialog.dispatchEvent(new CustomEvent('w-dialog:hide'));

      expect(dialog.getAttribute('aria-hidden')).toEqual('true');
    });
  });
});
```


### 5. Teleport Controller & tests

* Remove the template content by default, this is not super critical but is a nice thing to do to clean up, also aligns with the broad goals of 'teleporting' an element. Allow this to be overridden by `persist`.
* Instead of defaulting the target selector to `'body'`, let's allow this to work a different way and if we are in the shadow DOM (i.e. userbar), append to the root element within that DOM instead of the body.
* Use `get` methods to make the class a bit nicer to work with, add some refined error handling.
* Starting unit test - hope this helps.

```ts
import { Controller } from '@hotwired/stimulus';

/**
 * Allows the controlled element's content to be copied and appended
 * to another place in the DOM. Once copied, the original controlled
 * element will be removed from the DOM.
 * If no target selector is provided it will either append to the body
 * or to the Shadow Root's first DOM node depending on location of
 * the controlled element.
 *
 * @example
 * <template
 *  data-controller="w-teleport"
 *  data-w-teleport-target-value="dialog_root_selector"
 * >
 *  <div>Some content</div>
 * </template>
 */
export class TeleportController extends Controller<HTMLTemplateElement> {
  static values = {
    persist: { default: false, type: Boolean },
    target: { default: '', type: String },
  };

  declare persistValue: boolean;
  declare targetValue: string;

  connect() {
    this.append();
  }

  append() {
    const targetContainer = this.targetContainer;
    targetContainer.append(this.templateContent);
    if (this.persistValue) return;
    this.element.remove();
  }

  /**
   * Resolve a valid target container, defaulting to the document.body
   * or the shadow root's first DOM node if no target selector provided.
   */
  get targetContainer() {
    let targetContainer: any;

    if (this.targetValue) {
      targetContainer = document.querySelector(this.targetValue);
    } else {
      const rootNode = this.element.getRootNode();
      targetContainer =
        rootNode instanceof Document ? rootNode.body : rootNode.firstChild;
    }

    if (!(targetContainer instanceof Element)) {
      throw new Error(
        `No valid target container found at ${
          this.targetValue ? `'${this.targetValue}'` : 'the root node'
        }.`,
      );
    }

    return targetContainer;
  }

  get templateContent() {
    const templateContent =
      this.element.content.firstElementChild?.cloneNode(true);

    if (!(templateContent instanceof HTMLElement)) {
      throw new Error('Invalid template content.');
    }

    return templateContent;
  }
}
```

```ts
import { Application } from '@hotwired/stimulus';

import { TeleportController } from './TeleportController';

describe('TeleportController', () => {
  let application;

  describe('basic behaviour', () => {
    beforeEach(() => {
      application?.stop();

      document.body.innerHTML = `
  <main>
    <template id="template" data-controller="w-teleport">
      <div id="content">Some content</div>
    </template>
  </main>`;

      application = new Application();
      application.register('w-teleport', TeleportController);
    });

    it('should move the Template element content to the body and remove the template by default', async () => {
      expect(document.querySelectorAll('template')).toHaveLength(1);
      expect(document.getElementById('content')).toBeNull();

      application.start();

      await Promise.resolve();

      expect(document.querySelectorAll('template')).toHaveLength(0);
      expect(document.getElementById('content')).not.toBeNull();
      expect(document.getElementById('content').parentElement).toEqual(
        document.body,
      );
    });

    it('should allow a value to have the Template element persist', async () => {
      document
        .querySelector('template')
        .setAttribute('data-w-teleport-persist-value', 'true');

      expect(document.querySelectorAll('template')).toHaveLength(1);
      expect(document.getElementById('content')).toBeNull();

      application.start();

      await Promise.resolve();

      expect(document.querySelectorAll('template')).toHaveLength(1);
      expect(document.getElementById('content')).not.toBeNull();
      expect(document.getElementById('content').parentElement).toEqual(
        document.body,
      );
    });

    it.todo(
      'should allow the target container to be based on a provided selector value',
    );

    it.todo(
      'should allow for a default target container within the root element of a shadow DOM',
    );

    it.todo('should throw an error if the template content is empty');

    it.todo(
      'should throw an error if a valid target container cannot be resolved',
    );
  });
});
```


### 6. `client/src/includes/userbar.ts`

```ts
import axe, { ElementContext, NodeResult, Result, RunOptions } from 'axe-core';
import type A11yDialog from 'a11y-dialog';

import { Application } from '@hotwired/stimulus';
import { DialogController } from '../controllers/DialogController';
import { TeleportController } from '../controllers/TeleportController';

/*
This entrypoint is not bundled with any polyfills to keep it as light as possible
/// .....

// line ~57 - note the cloding of the firstElementChild (as per MDN guidelines for cloning, now that we also have the one root div in the `template`).
    shadowRoot.appendChild(
      (template.content.firstElementChild as HTMLElement).cloneNode(true),
    );


// line ~370 to ~402 - refactor this to create a Stimulus application and then use the Dialog event listener to get the body target.

    if (results.violations.length) {
      const a11yErrorBadge = document.createElement('span');
      a11yErrorBadge.textContent = String(a11yErrorsNumber);
      a11yErrorBadge.classList.add('w-userbar-axe-count');
      this.trigger.appendChild(a11yErrorBadge);
    }

    const stimulus = Application.start(
      this.shadowRoot.firstElementChild as Element,
    );

    stimulus.debug = process.env.NODE_ENV === 'development';

    stimulus.register('w-dialog', DialogController);
    stimulus.register('w-teleport', TeleportController);

    const modalReady = new Promise<{ body: HTMLElement; dialog: A11yDialog }>(
      (resolve) => {
        this.shadowRoot?.addEventListener('w-dialog:ready', (({
          detail,
        }: CustomEvent<{ body: HTMLElement; dialog: A11yDialog }>) => {
          resolve(detail);
        }) as EventListener);
      },
    );

    stimulus.start();

    const { body: modalBody, dialog: modal } = await modalReady;

    const accessibilityResultsBox = this.shadowRoot.querySelector(
      '#accessibility-results',
    );

```

**Comment by lb-** [comment]
@Lovelyfin00 how are you going with this, do you think you would still be keen to continue on this?

**Comment by Lovelyfin00** [comment]
Hey @lb- 
Yes I want to still work on it
I’m sorry but I’ve been so so busy I haven’t even had time for myself 😩😩

I finished the changes requested but I got stuck on the userbar.ts type changes
I’ll push them later today and ask for help on slack

**Comment by lb-** [comment]
Awesome, thanks Loveth! Happy to help with the TS stuff.

**Comment by Lovelyfin00** [comment]
> Awesome, thanks Loveth! Happy to help with the TS stuff.

Hey @lb- 
I've been able to resolve it
I'm including the tests now
Thanks😅😅

**Comment by lb-** [comment]
Awesome @Lovelyfin00 ! - mark as ready when it's good for another review and ping me, I will do my best to get to it this week.

**Comment by Lovelyfin00** [comment]
Hey @lb- 
I have issues with setting the theme value
It keeps returning an empty string even after I set it in the test

File: client/src/controllers/DialogController.test.js [hunk 4]
```diff
@@ -0,0 +1,141 @@
...
+            dialog.dispatchEvent(new CustomEvent('w-dialog:hide'));
+
+            expect(dialog.getAttribute('aria-hidden')).toEqual('true');
+
+            dialog.dispatchEvent(new CustomEvent('w-dialog:show'));
+
+            // adding a theme value to remove the default behaviour
+            dialog.setAttribute("class", "w-dialog--floating");
+            dialog.setAttribute("data-w-dialog-theme-value", "floating");
+
```
Comment (lb-) [inline client/src/controllers/DialogController.test.js:130]: When making a data value change, you may need to ensure that all DOM behaviour has resolved.

Try ...

```
await Promise.resolve();
```

Comment (lb-) [reply client/src/controllers/DialogController.test.js:130]: Also, when you set this value, the dialog code has already run.

You will need to set the value before you do application.start.

File: client/src/includes/userbar.ts [hunk 5]
```diff
@@ -51,6 +51,7 @@ export class Userbar extends HTMLElement {
       '#wagtail-userbar-template',
     );
     if (!template) return;
+    console.log(this);
```
Comment (zerolab) [inline client/src/includes/userbar.ts:54]: rogue console.log?

Comment (Lovelyfin00) [reply client/src/includes/userbar.ts:54]: Yeah😅😅😅
And it was caught by the linter 😅

**Comment by Lovelyfin00** [comment]
Hey @lb- 
I won’t be able to finish up with the PR
I’m traveling for my one year service and gonna be off the internet for three weeks
It’s remaining just the TeleportController test for the PR to be ready😅😅

**Comment by lb-** [comment]
Thanks @Lovelyfin00 I'll give this another manual review and see if I can just add that last test case myself. 

Really appreciate you working through this.

**Review by lb- — APPROVED** [review]
@Lovelyfin00  this works great, thank you so much for following through on all of this.

I have made some refinements and will rebase/merge this in shortly.
Again, I really appreciate you taking the time to do this!

- Added event dispatching to the TeleportController
- Updated JSDoc on both new controllers, just refined examples & wording
- Added final test cases for TeleportController (it was a bit tricky but will serve as a nice way to test error handling in Stimulus for future work)
- Added upgrade considerations (for event name changes) and also added a comment to the dialog template so we remove the legacy events in a future release
- Changed the userbar wrapper to an `aside` element
- Ensure that the dialog controller will dispatch the shown/hidden events even when theme is floating.
- Fix up tests / linting items
- Changed TeleportController value from `persist` to `keep` - shorter and seemed simpler
- Fix issue in userbar usage where application was being started twice
- Change the userbar event listener to be `once` so that we do not need to clean up any event listeners in our code

**Comment by lb-** [comment]
Curlylint did not seem to like the Django comment in between html attributes.

**Pull request merged**